src/client/components/terminal.ts

```diff
@@ -373,6 +394,50 @@ export class TerminalView extends LitElement {
     }
   }
 
+  #resizeTerminal(rows?: number) {
+    if (rows !== undefined) { this.rows = rows }
+    return this.fitAddon?.fit(this.rows)
+  }
+
+  #createResizeHandle(): HTMLElement {
+    const dragHandle = document.createElement('div')
+    dragHandle.setAttribute('class', 'xterm-drag-handle')
+
+    let dragState: {
+      initialClientY: number
+      initialRows: number
+    } | undefined
+
+    dragHandle.addEventListener('mousedown', (e) => {
+      dragState = {
+        initialClientY: e.clientY,
+        initialRows: this.rows
+      }
+    })
+
+    window.addEventListener('mouseup', () => {
+      if (dragState === undefined) { return }
+      dragState = undefined
+    })
+
+    window.addEventListener('mousemove', (e) => {
+      if (dragState === undefined || !this.fitAddon) { return }
+
+      const delta = e.clientY - dragState.initialClientY
+
+      console.log({ delta, cellHeight: this.fitAddon.getCellSize().height })
+
+      const deltaRows = delta / this.fitAddon.getCellSize().height
+      const newRows = Math.round(dragState.initialRows + deltaRows)
+
+      if (newRows !== this.rows) {
+        this.#resizeTerminal(newRows)
```

Comment:
You've mentioned that resizing the terminal will not focus, but there is any reason `this.terminal?.focus()` won't work here? I just tested with that change, and the terminal focus is kept after resizing.

Reply:
OH yeah, I definitely should've tried that. Fixed, thanks!